cmd/nerdctl/pull.go

```diff
@@ -48,6 +54,7 @@ func newPullCommand() *cobra.Command {
 	pullCommand.Flags().StringSlice("platform", nil, "Pull content for a specific platform")
 	pullCommand.RegisterFlagCompletionFunc("platform", shellCompletePlatforms)
 	pullCommand.Flags().Bool("all-platforms", false, "Pull content for all platforms")
+	pullCommand.Flags().Bool("verify", false, "Verify the image with cosign")
```

Comment:
Maybe this should be a string flag `--verify=(none|cosign)` so that we can eventually add other verifier such as Notary (v2).

Reply:
Also, can we have integration tests?
The tests should cover both good sign and bad sign.

Reply:
String flag `--verify=(none|cosign)` would make sense to implement! However, what about adding this feature when someone wants to work with notary to implement in nerdctl? But we can do it anyway if you want, since it's a good long-term feature!

For the integration tests, we probably need to store `cosign.key` and `cosign.pub` files in the `./examples` or smth folder. And we will be depended on `cosign` executable (must be installed) to run all integration tests, is it OK?

Reply:
Notary support doesn't need to be added right now. We just have to design the CLI flag to be extensible to cover notary (or something else) in future.

> For the integration tests, we probably need to store cosign.key and cosign.pub files in the ./examples or smth folder. And we will be depended on cosign executable (must be installed) to run all integration tests, is it OK?

The key should be generated during the tests by executing `cosign`. (as we generate ocicrypt keys in `image_encrypt_linux_test.go`)

Reply:
(So, maybe we should change `push --cosign` to `push --sign=(none|cosign)` for consistency with `pull --verify=(none|cosign)`. Sorry for going back and forth.)

Reply:
Great, we've enhanced the flag according to your reviews

Reply:
I think we [wrote a simple integration test](https://github.com/containerd/nerdctl/pull/556/commits/2bd34d2add552cad98345edb94d887865bffec13) that creates a cosign-key-pair and run push&pull test cases, but not sure if it is effective. We couldn't able to run the tests in Vagrant since it could not find the `testutil.go` file. 🤔 What is the right way to run the all test?